cli/flox/src/commands/mod.rs

```diff
@@ -516,25 +516,39 @@ impl UpdateNotification {
             Err(e) => Err(UpdateNotificationError::Io(e))?,
         };
 
-        let new_version = get_latest_version_future.await?;
+        let new_version_str = get_latest_version_future.await?;
+        let current_version_str = FLOX_VERSION.as_str();
 
-        // Sanity check we got a version back
-        if let Err(e) = semver::Version::parse(&new_version) {
-            return Err(UpdateNotificationError::WeMayHaveMessedUp(anyhow!(
-                "version is invalid: {e}"
-            )));
+        if let Ok(new_version) = new_version_str.parse::<FloxVersion>() {
+            if let Ok(current_version) = current_version_str.parse::<FloxVersion>() {
```

Comment:
note: I'd have to think about how we could do it, but it would be nice to check that FLOX_VERSION parses at some point during our build/test/release process

Reply:
If we change that `FLOX_VERSION` would be of type `FloxVersion` then running `flox --version` would do the `.parse() -> .to_string()` and check that the format is valid.

We can than run `flox --version` in release pipeline (we already do) and we can run this in the `postInstallCheck` in `pkgs/flox-cli/default.nix`

What do you think?

Reply:
Sounds great! But we `--set FLOX_VERSION` in `pkgs/flox/default.nix`, so would we need to do it there rather than in `flox-cli`?